balancer-js/examples/pools/queries.ts

```diff
@@ -1,62 +1,73 @@
 /**
  * Shows how to query balancer helper contracts for
  * expected amounts when providing or exiting liquidity from pools
- * 
+ *
  * yarn examples:run ./examples/pools/queries.ts
  */
 
-import { parseEther, formatEther } from '@ethersproject/units'
-import { BalancerSDK, PoolWithMethods } from '@/.'
+import { parseEther, formatEther } from '@ethersproject/units';
+import { BalancerSDK, PoolWithMethods } from '@/.';
 
 const sdk = new BalancerSDK({
   network: 1,
   rpcUrl: 'https://eth-rpc.gateway.pokt.network',
-})
+});
 
 const { pools, balancerContracts: contracts } = sdk;
 
 // Joining with a single token
 const queryJoin = async (pool: PoolWithMethods) => {
-  const token = pool.tokensList[0]
+  const token = pool.tokensList[0];
   const joinExactInQuery = pool.buildQueryJoinExactIn({
-    maxAmountsIn: pool.tokensList.map((t) => parseEther((t === token) ? '1' : '0'))
-  })
+    maxAmountsIn: pool.tokensList.map((t) =>
+      parseEther(t === token ? '1' : '0')
+    ),
+  });
 
-  const response = await contracts.balancerHelpers.queryJoin(...joinExactInQuery)
+  const response = await contracts.balancerHelpers.callStatic.queryJoin(
+    ...joinExactInQuery
+  );
 
-  console.log(`Joining ${pool.poolType}`)
+  console.log(`Joining ${pool.poolType}`);
   console.table({
     tokens: pool.tokensList.map((t) => `${t.slice(0, 6)}...${t.slice(38, 42)}`),
     amountsIn: response.amountsIn.map(formatEther),
     bptOut: formatEther(response.bptOut),
-  });  
-}
+  });
+};
 
 // Exiting to single token
 const queryExit = async (pool: PoolWithMethods) => {
   const exitToSingleToken = pool.buildQueryExitToSingleToken({
     bptIn: parseEther('1'),
-    tokenOut: pool.tokensList[0]
-  })
+    tokenOut: pool.tokensList[0],
+  });
 
-  const response = await contracts.balancerHelpers.queryExit(...exitToSingleToken)
+  const response = await contracts.balancerHelpers.callStatic.queryExit(
```

Comment:
Why did you have to add `callStatic` to these method calls? I was under the impression it would be an "exact match" replace, so I'm trying to understand what's changing here.

Reply:
See the docs in IBalancerQueries in monorepo:
```
 * If calling them from an off-chain client, make sure to use eth_call: most clients default to eth_sendTransaction for
 * non-view functions.
 ```
 Ethers callStatic use eth_call behind the scenes. I thinking the changes I made the typing more stict and the previous code was throwing an error.